Examples/Widgets/Timer/src/components/App.ts

```diff
@@ -6,51 +6,82 @@
  *
  */
 
-import { DateTime, Duration } from "luxon";
-import { html, LitElement, customElement, internalProperty, property, query, PropertyValues } from "lit-element";
+import { DateTime, Duration, Interval } from "luxon";
+import { html, LitElement, customElement, internalProperty, property, query } from "lit-element";
 import styles from "./App.scss";
 import "./time-circle"
 @customElement("timer-component")
 export default class TimerComponent extends LitElement {
   @property({ type: String, reflect: true }) duration = ""
+  @property({ type: String, reflect: true }) remaining = ""
   @property({ type: Boolean }) paused = false
   @property({ type: String, attribute: "hours-color" }) hoursColor = "#0A78CC"
   @property({ type: String, attribute: "minutes-color" }) minutesColor = "#73A321"
   @property({ type: String, attribute: "seconds-color" }) secondsColor = "#875AE0"
 
+  @internalProperty() endTime = "";
   @internalProperty() durationData!: Duration;
-  @internalProperty() remaining!: string;
   @internalProperty() remainingHours!: string;
   @internalProperty() remainingMinutes!: string;
   @internalProperty() remainingSeconds!: string;
 
   @query(".container") container!: HTMLElement;
 
+  get storedStartTime() {
+    return window.localStorage.getItem("timer-start") || DateTime.local().toString()
+  }
+  get storedDuration() {
+    return window.localStorage.getItem("timer-duration") || this.duration
+  }
+
   connectedCallback() {
     super.connectedCallback();
-    const savedTime = window.localStorage.getItem("time-remaining") ? window.localStorage.getItem("time-remaining")! : this.duration;
-    this.resetDuration(savedTime)
+    this.setDuration(this.storedStartTime, this.storedDuration)
     setInterval(() => {
       if (!this.paused) {
         this.durationData = this.durationData.minus({ seconds: 1 })
         this.remaining = this.durationData.toFormat('hh:mm:ss');
-        this.remainingHours = this.remaining.split(":")[0]
-        this.remainingMinutes = this.remaining.split(":")[1]
-        this.remainingSeconds = this.remaining.split(":")[2]
-        this.duration = this.remaining
-        window.localStorage.setItem("time-remaining", this.remaining)
+        let [hours, minutes, seconds] = this.durationData.toFormat('hh:mm:ss').split(":");
+        this.remainingHours = hours
```

Comment:
There are all of semi colons missing. Not sure if this is something we care about or not. I'm just not used to seeing it.

Reply:
@elenagiere hm yeah . . . I mean it's all valid JavaScript, so if we want to add a style-guide and lint rules, that might be a next-level advancement for the whole starter library.